Fix k8s helm chart support for prefixed deployments

jcrist opened the pull request:
Fixes support for deploying dask-gateway on k8s with all routes served
under a prefix (e.g. `/services/dask-gateway/`).  This was woefully
broken on master, and due to lack of tests was missed.

Fixes #239

jcrist commented:
cc @consideratio, this should fix your issue.

consideRatio commented on resources/helm/dask-gateway/templates/gateway/middleware.yaml:
```diff
@@ -0,0 +1,23 @@
+spec:
+  stripPrefixRegex:
+    regex:
+      - '{{ .Values.gateway.prefix | trimSuffix "/" }}/clusters/[a-zA-Z0-9.-]+/'
```
This middleware will not work if someone calls `/clusters/name1`, but only if someone calls `/clusters/name1/`. This is a small difference that I've recently run into issues with for other software.

I'm considering this:

```suggestion
      - '{{ .Values.gateway.prefix | trimSuffix "/" }}/clusters/[a-zA-Z0-9.-]+'
```

And if we should make things more explicit by setting this: https://docs.traefik.io/middlewares/stripprefix/#forceslash

consideRatio commented on resources/helm/dask-gateway/templates/gateway/middleware.yaml:
```diff
@@ -0,0 +1,23 @@
+  labels:
+    {{- include "dask-gateway.labels" . | nindent 4 }}
+spec:
+  stripPrefixRegex:
```
```suggestion
  stripPrefix:
```

consideRatio commented on resources/helm/dask-gateway/templates/gateway/middleware.yaml:
```diff
@@ -0,0 +1,23 @@
+spec:
+  stripPrefixRegex:
+    regex:
+      - '{{ .Values.gateway.prefix | trimSuffix "/" }}/'
```
Similarly as above, I consider:

```suggestion
      - '{{ .Values.gateway.prefix | trimSuffix "/" }}'
```

jcrist commented:
Thanks for the comments, updated. Note that none of the backends (api server or cluster dashboards) support a root route, so forcing a trailing slash isn't a problem here. Removed it anyway since it causes no harm.

consideRatio commented:
Ah! A concern I had was that the rule would not trigger on a request that didn't include the trailing `/`.

Thanks for your quick work to resolve this!

jcrist commented:
If you're able to deploy from master, I'm going to try to fix your other issue and push a patch release in the next few days. Please let me know if you run into any other problems.
